### PR TITLE
langgraph-cli: Add langgraph-cli formula

### PR DESCRIPTION
**Pull request by hinthornw**
This PR adds installer for the LangGraph CLI, a small package for building and deploying applications to the LangGraph platform.

License: MIT
Pypi: https://pypi.org/project/langgraph-cli/
Homepage: https://github.com/langchain-ai/langgraph/tree/main/libs/cli

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by hinthornw** [comment]
The error seems unrelated to my PR.

the error:
```
Full audit --except=installed --tap=homebrew/core output
  hasura-cli
    * Dependency 'node@18' is deprecated but has un-deprecated dependents. Either
      un-deprecate 'node@18' or deprecate it and all of its dependents.
```

I'm not touching the  `hasura-cli` in any way here.

File: Formula/l/langgraph-cli.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class LanggraphCli < Formula
+  include Language::Python::Virtualenv
+
+  desc "Shiny new formula"
```
Comment (carlocab) [inline Formula/l/langgraph-cli.rb:4]: Could you choose a better description?

File: Formula/l/langgraph-cli.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    url "https://files.pythonhosted.org/packages/96/d3/f04c7bfcf5c1862a2a5b845c6b2b360488cf47af55dfa79c98f6a6bf98b5/click-8.1.7.tar.gz"
+    sha256 "ca9853ad459e787e2192211578cc907e7594e294c7ccc834310722b41b9ca6de"
+  end
+
+  def python3
+    "python3.13"
+  end
+
+  def install
+    virtualenv_create(libexec, python3)
```
Comment (carlocab) [inline Formula/l/langgraph-cli.rb:22]:
```suggestion
  def install
```
You don't need this.

File: Formula/l/langgraph-cli.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "Usage: langgraph", shell_output("#{bin}/langgraph --help")
```
Comment (carlocab) [inline Formula/l/langgraph-cli.rb:27]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for the PR. A few comments.

File: Formula/l/langgraph-cli.rb (hunk 3 above)
Comment (hinthornw) [reply Formula/l/langgraph-cli.rb]: Yeah! Thank you for the review! Was trying to get the rebase working and overwrote my other tests & descriptions - will push a new one up - apologies!

**Pull request closed without merging**